dance/modules/multi_modality/match_modality/cmae.py

```diff
@@ -171,7 +171,11 @@ def forward(self, images):
 
     def encode(self, images):
         hiddens = self.enc(images)
-        noise = Variable(torch.randn(hiddens.size()).cuda(hiddens.data.get_device()))
+        device_index = hiddens.data.get_device()
+        if device_index < 0:
+            noise = Variable(torch.randn(hiddens.size()))
```

Comment:
1) Do not use Variable, as it is deprecated (https://pytorch.org/docs/stable/autograd.html#variable-deprecated)
2) Use `randn_like` with `hiddens`, which should remove the need to manually infer device (https://pytorch.org/docs/stable/generated/torch.randn_like.html)

Reply:
Yes,thanks,I have changed it